src/assignment_logger.rs

```diff
@@ -0,0 +1,29 @@
+use serde::{Deserialize, Serialize};
+
+use crate::SubjectAttributes;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AssignmentEvent {
+    pub experiment: String,
+    pub variation: String,
+    pub subject: String,
+    pub subject_attributes: SubjectAttributes,
+    pub timestamp: String,
+    pub feature_flag: String,
+    pub allocation: String,
+}
```

Comment:
This definition is almost the desired shape just missing the `metaData` field which is a generic container. 

In the python implementation is contains:

```
"metaData": {"sdkLanguage": "python", "sdkVersion": __version__},
```

No problem! I apologize if the design spec didn't make clear but I  wanted to mention again that the UFC migration code in the javascript and python code is not yet merged to main. 

Two SDKs have been converted to the forthcoming “UFC” format are:

* https://github.com/Eppo-exp/python-sdk/pull/28
* https://github.com/Eppo-exp/js-client-sdk-common/pull/45

Sorry again if this caused confusion - they should be merged to main shortly to make reading them easier.

Reply:
Good catch! I added this field and also added missing `camelCase` transformation